_ingest-pipelines/processors/date-index-name.md

```diff
@@ -0,0 +1,154 @@
+`index_name_prefix` | Optional | The index name prefix to add before the date. Supports [template snippets]({site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`description`  | Optional  | A brief description of the processor.  |
+`if` | Optional | A condition for running this processor. |
+`ignore_failure` | Optional | If set to `true`, failures are ignored. Default is `false`. |
+`locale` | `locale`  | Optional  | The locale to use when parsing the date. Default is `ENGLISH`. Supports [template snippets]({site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).  |
+`on_failure` | Optional | A list of processors to run if the processor fails. |
+`tag` | Optional | An identifier tag for the processor. Useful for debugging to distinguish between processors of the same type. |
+`timezone`  | Optional  | The time zone to use when parsing the date. Default is `UTC`. Supports [template snippets]({site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+**Step 1: Create a pipeline.**
+
+The following query creates a pipeline, named `date-index-name1`, that uses the `date_index_name` processor to index logs into monthly indexes: 
+
+```json
+PUT /_ingest/pipeline/date-index-name1
+{
+  "description": "Create weekly index pipeline",
+  "processors": [
+    {
+      "date_index_name": {
+        "field": "date_field",
+        "index_name_prefix": "week_index-",
+        "date_rounding": "w",
+        "date_formats": ["YYYY-MM-DD"]
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+**Step 2 (Optional): Test the pipeline.**
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/date-index-name1/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "date_field": "2023-10-30"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "<week_index-{2023-10-01||/w{yyyy-MM-dd|UTC}}>",
+        "_id": "1",
+        "_source": {
+          "date_field": "2023-10-30"
+        },
+        "_ingest": {
+          "timestamp": "2023-11-13T18:23:10.408593092Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+**Step 3: Ingest a document.**
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=date-index-name1
+{
+  "date_field": "2023-10-30"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request indexes the document into the index `week_index-2023-10-23` and will continue to index documents into weekly indexes because the pipeline rounds by week.
```

Comment:
```suggestion
The request indexes the document into the index `week_index-2023-10-23` and will index all documents with a timestamp within that week into the same index because the pipeline rounds by week.
```